TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java

```diff
@@ -380,6 +376,12 @@ public void onTangoStatus(int status) {
             saveUuidsNamestoHashMap();
             mParameterNode.setPreferencesFromParameterServer();
             mMapSaved = false;
+            if (mParameterNode.getBoolParam(getString(R.string.pref_create_new_map_key)) ||
+                    mParameterNode.getIntParam(getString(R.string.pref_localization_mode_key)) == 3)
+                getTangoPermission(EXTRA_VALUE_ADF, REQUEST_CODE_ADF_PERMISSION);
+            if (!mParameterNode.getStringParam(getString(R.string.dataset_datasets_path_key)).isEmpty() &&
+                    !mParameterNode.getStringParam(getString(R.string.dataset_uuid_key)).isEmpty())
+                getTangoPermission(EXTRA_VALUE_DATASET, REQUEST_CODE_DATASET_PERMISSION);
```

Comment:
The documentation is not really clear about it but I think not, they are required only for specific Tango API calls like: `TangoService_getAreaDescriptionUUIDList`, `TangoService_getAreaDescriptionMetadata`, `TangoService_saveAreaDescription` or `TangoService_Experimental_getDatasetUUIDs`.

What we could also do is, each time we do one of this call from the C++ code, somehow trigger this intent in the Java code. I think it could be possible using a ros topic.